.github/workflows/lint.yaml

```diff
@@ -74,12 +74,18 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Run `go mod tidy`
-        run: rm go.sum api/go.sum && go mod tidy && (cd api && go mod tidy)
-
-      - name: Check for no changes after `go mod tidy`
-        # We have to add the current directory as a safe directory or else git commands will not work as expected.
-        run: git config --global --add safe.directory $( realpath . ) && git diff --exit-code -- go.mod go.sum api/go.mod api/go.sum
+      - name: Check for untidy go modules
+        shell: bash
+        run: |
+          find . -path ./e -prune -o -name go.mod -print | while read f; do 
```

Comment:
Internal tools has several projects not strictly tied to Teleport in this repo. It looks like if we manage to break one of the go.mod files for one of these, then this check will fail for product PRs until our team fixes the issue. Is this intentional?

Longer term, most if not all of these tools need to move to shared-workflows/similar repo.

Reply:
If a go.mod file that fails this checked gets committed to teleport.e then it will block `e` ref updates on OSS teleport repo

Reply:
Ah sorry I misread the `find` command. Please disregard.